app/src/sharedTest/java/org/oppia/app/walkthrough/WalkthroughActivityTest.kt

```diff
@@ -28,18 +36,28 @@ class OnboardingFragmentTest {
   @Test
   fun testWalkthroughFragment_increaseProgress_worksCorrectly() {
     ActivityScenario.launch(WalkthroughActivity::class.java).use {
-      onView(withId(R.id.next_button)).perform(click())
+      onView(withId(R.id.walkthrough_welcome_next_btn)).perform(click())
+      onView(withId(R.id.walkthrough_progress_bar)).check(matches(withProgress(2)))
+    }
+  }
+
+  @Test
+  fun testWalkthroughFragment_increaseProgress_onBackPressed_decreaseProgress_progressWorksCorrectly() {
+    ActivityScenario.launch(WalkthroughActivity::class.java).use {
+      onView(withId(R.id.walkthrough_welcome_next_btn)).perform(click())
+      onView(withId(R.id.walkthrough_progress_bar)).check(matches(withProgress(2)))
+      pressBack()
       onView(withId(R.id.walkthrough_progress_bar)).check(matches(withProgress(1)))
     }
   }
 
   @Test
   fun testWalkthroughFragment_increaseProgress_decreaseProgress_progressWorksCorrectly() {
     ActivityScenario.launch(WalkthroughActivity::class.java).use {
-      onView(withId(R.id.next_button)).perform(click())
-      onView(withId(R.id.walkthrough_progress_bar)).check(matches(withProgress(1)))
+      onView(withId(R.id.walkthrough_welcome_next_btn)).perform(click())
```

Comment:
@nikitamarysolomanpvt this will he sufficient to check whether page progress align with progress bar

Reply:
Ya this will do please ignore that comment